### PR TITLE
Add error handling and error layout to Cyberstorm

### PR DESCRIPTION
**Pull request by VilppeRiskidev**

### CODE REVIEW

**Comment by VilppeRiskidev** [comment]
Examples (when running nextjs): `http://localhost:3000/error/500` `http://localhost:3000/error/404` `http://localhost:3000/error/1234` `http://localhost:3000/abc`

File: apps/cyberstorm-nextjs/app/error/[error]/page.tsx
Comment (anttimaki) [inline apps/cyberstorm-nextjs/app/error/[error]/page.tsx:1]: I guess this is the test page Mythic asked you to remove?

File: packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import styles from "./ErrorLayout.module.css";
+import { BaseLayout } from "../BaseLayout/BaseLayout";
+
+export interface ErrorLayoutProps {
+  error: 404 | 500;
+}
+
+/**
+ * Cyberstorm NotFound Layout
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx:9]: Copy-pasted inaccurate comment.

File: packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const { error } = props;
+
+  let errorCode = "Error";
+  let errorDescription = "Unknown";
+  let errorFlavorText = "An error with an unknown error code happened";
+
+  if (error === 404) {
+    errorCode = "404";
+    errorDescription = "Page not found";
+    errorDescription = "Page not found";
```
Comment (anttimaki) [inline packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx:21]: I got it the first time :trollface:

**Review by anttimaki — COMMENTED** [review]
I'm assuming logging the errors to Sentry is outside the scope of the PR?

Comments on the code itself are low prio. However, testing locally I encountered odd behaviour:

1. Open the site (homepage)
2. Navigate to Developers -> Wiki to get a 404.
3. Navigate e.g. to Communities. The URL changes, but 404 page's contents are still shown

Not sure what happens there, and at least once the whole things worked as expected. But probably worth checking out if this is just me or a prevalent problem.

**Comment by MythicManiac** [comment]
> I'm assuming logging the errors to Sentry is outside the scope of the PR?

Yes

File: packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx (hunk 2 above)
Comment (VilppeRiskidev) [reply packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx]: Ctrl + D and Ctrl + S are dangerously close to each other xD

**Review by anttimaki — APPROVED** [review]

File: packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx [hunk 3]
```diff
@@ -2,7 +2,13 @@ import styles from "./ErrorLayout.module.css";
 import { BaseLayout } from "../BaseLayout/BaseLayout";
 
 export interface ErrorLayoutProps {
-  error: 404 | 500;
+  error: 404 | 500 | number;
```
[the remark this answers is not in the document]
Comment (MythicManiac) [reply packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx:5]: On the other hand, it makes it a bit more clear that at the very least those two are known cases. IMO the `number` could be removed if it's not used yet however.

[the remark this answers is not in the document]
Comment (VilppeRiskidev) [reply packages/cyberstorm/src/components/Layout/ErrorLayout/ErrorLayout.tsx]: removed `number` and the unknown error handling related to it.

File: apps/cyberstorm-nextjs/app/error.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import { useEffect } from "react";
+import { ErrorLayout } from "@thunderstore/cyberstorm";
+
+export default function Error({ error }: { error: Error; reset: () => void }) {
+  useEffect(() => {
+    console.error(error);
+  }, [error]);
+
+  return <ErrorLayout error={500} />;
+}
```
Comment (MythicManiac) [inline apps/cyberstorm-nextjs/app/error.tsx:12]: This component should not have `"use client"` nor should it use state for anything.

**Pull request merged**